crates/apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -308,192 +297,54 @@ where
                     None
                 };
             let tx_gas_meter = RefCell::new(tx_gas_meter);
-            let tx_result = protocol::dispatch_tx(
-                tx.clone(),
+            let mut tx_event = new_tx_event(&tx, height.0);
+            let is_atomic_batch = tx.header.atomic;
+            let commitments_len = tx.commitments().len() as u64;
+            let tx_hash = tx.header_hash();
+            let mut tx_result = namada::tx::data::TxResult::default();
```

Comment:
@tzemanovic any suggestions here? Because we might need to commit some txs of the batch even in case of an error I need to always have a `TxResult` to analyze. Here I modified `dispatch_tx` to take it as an argument instead of returning it. I thought about embedding the `TxResult` even inside the error type but that might load that type too much

Reply:
personally I'd still put it in the return type

Reply:
Tried to fix it in db8fa3625e194be1a4f2d367ed358ff4fa6d6bd4

Reply:
looks good, thx!